pkg/restapi/issuer/operation/operations_test.go

```diff
@@ -127,6 +127,130 @@ func TestOperation_Callback(t *testing.T) {
 	require.Contains(t, body.String(), "unable to load html")
 }
 
+func TestOperation_RetrieveVC(t *testing.T) {
+	headers := make(map[string]string)
+	headers["Authorization"] = authHeader
+
+	file, err := ioutil.TempFile("", "*.html")
+	require.NoError(t, err)
+
+	defer func() { require.NoError(t, os.Remove(file.Name())) }()
+
+	t.Run("retrieve credential success", func(t *testing.T) {
+		router := mux.NewRouter()
+
+		router.HandleFunc("/retrieve", func(writer http.ResponseWriter, request *http.Request) {
+			writer.WriteHeader(http.StatusOK)
+			_, err := writer.Write([]byte(`"credential"`))
+			if err != nil {
+				panic(err)
+			}
+		})
+
+		vcs := httptest.NewServer(router)
+
+		defer vcs.Close()
+
+		cfg := &Config{TokenIssuer: &mockTokenIssuer{}, TokenResolver: &mockTokenResolver{},
+			VCSURL: vcs.URL, ReceiveVCHTML: file.Name(), QRCodeHTML: file.Name()}
+
+		handler := getHandlerWithConfig(t, retrieve, cfg)
+		_, status, err := handleRequest(handler, headers, retrieve)
+		require.NoError(t, err)
+		require.Equal(t, http.StatusOK, status)
+	})
+	t.Run("retrieve credential error", func(t *testing.T) {
+		router := mux.NewRouter()
+
+		router.HandleFunc("/retrieve", func(writer http.ResponseWriter, request *http.Request) {
+			writer.WriteHeader(http.StatusInternalServerError)
+		})
+
+		vcs := httptest.NewServer(router)
+
+		defer vcs.Close()
+
+		cfg := &Config{TokenIssuer: &mockTokenIssuer{}, TokenResolver: &mockTokenResolver{},
+			VCSURL: vcs.URL, ReceiveVCHTML: file.Name(), QRCodeHTML: file.Name()}
+		handler := getHandlerWithConfig(t, retrieve, cfg)
+
+		_, status, err := handleRequest(handler, headers, retrieve)
+		require.NoError(t, err)
+		require.Equal(t, http.StatusInternalServerError, status)
+	})
+	t.Run("retrieve credential error", func(t *testing.T) {
```

Comment:
Redundant test?

Reply:
I will remove since this has to be removed once edv change gets pushed in